util/auto_tune_compressor.cc

```diff
@@ -144,138 +152,222 @@ CostAwareCompressor::CostAwareCompressor(const CompressionOptions& opts)
       continue;
     }
     if (kCompressionLevels[type - 1].size() == 0) {
-      allcompressors_.emplace_back();
       continue;
     } else {
-      // if the compression type is not supported, then skip and remove
-      // compression levels from the supported compression level list
+      // If the compression type is not supported, then skip and remove
+      // compression levels from the supported compression level list.
       if (std::find(compressions.begin(), compressions.end(), type) ==
           compressions.end()) {
-        allcompressors_.emplace_back();
         continue;
       }
-      std::vector<std::unique_ptr<Compressor>> compressors_diff_levels;
       for (size_t j = 0; j < kCompressionLevels[type - 1].size(); j++) {
         auto level = kCompressionLevels[type - 1][j];
         CompressionOptions new_opts = opts;
         new_opts.level = level;
-        compressors_diff_levels.push_back(
+        compressors_.emplace_back(
             builtInManager->GetCompressor(new_opts, type));
-        allcompressors_index_.emplace_back(i, j);
       }
-      allcompressors_.push_back(std::move(compressors_diff_levels));
     }
   }
+  MeasureUtilization();
+  block_count_ = 0;
+  cur_compressor_idx_ = 0;
+  default_compressor_ =
+      GetBuiltinV2CompressionManager()->GetCompressor(opts, default_type);
```

Comment:
No need for this. See https://github.com/facebook/rocksdb/pull/13769#discussion_r2211865587